cmd/aks-periscope/aks-periscope.go

```diff
@@ -55,35 +105,104 @@ func main() {
+	selectedDiagnosers := selectDiagnosers(
+		map[string]interfaces.Diagnoser{
+			networkConfigDiagnoser.GetName():   networkConfigDiagnoser,
+			networkOutboundDiagnoser.GetName(): networkOutboundDiagnoser,
+		})
+
+	//exporters
+	azureBlobExporter := exporter.NewAzureBlobExporter(creationTimeStamp, hostname)
+	selectedExporters := selectExporters(
+		map[string]interfaces.Exporter{
+			azureBlobExporter.GetName():    azureBlobExporter,
+		})
+
+	return selectedCollectors, selectedDiagnosers, selectedExporters
+}
+
+// selectCollectors select the collectors to run by looking up the value from the config
+func selectCollectors(allCollectorsByName map[string]interfaces.Collector) []interfaces.Collector {
+	collectors := []interfaces.Collector{}
+
+	enabledCollectorString, found := os.LookupEnv("ENABLED_COLLECTORS")
+	if !found {
+		//if not defined, default to all collectors enabled
+		enabledCollectorString = "containerlogs dns helm iptables kubeletcmd kubeobjects networkoutbound nodelogs osm smi systemlogs systemperf"
 	}
 
-	if contains(collectorList, "connectedCluster") {
-		collectors = append(collectors, helmCollector)
-	} else {
-		collectors = append(collectors, systemLogsCollector)
-		collectors = append(collectors, ipTablesCollector)
-		collectors = append(collectors, nodeLogsCollector)
-		collectors = append(collectors, kubeletCmdCollector)
-		collectors = append(collectors, systemPerfCollector)
+	enabledCollectorNames := strings.Fields(enabledCollectorString)
+
+	for _, collectorName := range enabledCollectorNames {
+		collectors = append(collectors, allCollectorsByName[collectorName])
 	}
 
-	// OSM and SMI flags are mutually exclusive
-	if contains(collectorList, "OSM") {
-		collectors = append(collectors, osmCollector)
-		collectors = append(collectors, smiCollector)
-	} else if contains(collectorList, "SMI") {
-		collectors = append(collectors, smiCollector)
+	return collectors
+}
+
+// selectDiagnosers select the diagnosers to run
+func selectDiagnosers(allDiagnosersByName map[string]interfaces.Diagnoser) []interfaces.Diagnoser {
+	diagnosers := []interfaces.Diagnoser{}
+
+	//read list of diagnosers that are enabled
+	enabledDiagnoserString, found := os.LookupEnv("ENABLED_DIAGNOSERS")
+	if !found {
+		//if not defined, default to all diagnosers enabled
+		enabledDiagnoserString = "networkconfig networkoutbound"
 	}
 
-	collectorGrp := new(sync.WaitGroup)
+	enabledDiagnoserNames := strings.Fields(enabledDiagnoserString)
+
+	for _, diagnoserName := range enabledDiagnoserNames {
+		diagnosers = append(diagnosers, allDiagnosersByName[diagnoserName])
+	}
+
+	return diagnosers
+}
+
+// selectedExporters select the exporters to run
+func selectExporters(allExporters map[string]interfaces.Exporter) []interfaces.Exporter {
+	exporters := []interfaces.Exporter{}
+
+	//read list of exporters that are enabled
+	enabledExportersString, found := os.LookupEnv("ENABLED_EXPORTERS")
```

Comment:
💡What will be associated `deployment.yaml` changes will look like? For this new var.

Reply:
https://github.com/davidkydd/aks-periscope/blob/0071201dd41cd159e87e5a574bdfbcc271c122b1/deployment/aks-periscope-arc.yaml#L163